src/transactions/RestoreFootprintOpFrame.cpp

```diff
@@ -47,180 +54,346 @@ RestoreFootprintOpFrame::RestoreFootprintOpFrame(
+            rustEntryRentChanges,
+            mSorobanConfig.rustBridgeRentFeeConfiguration(), ledgerSeq);
+        if (!mRefundableFeeTracker->consumeRefundableSorobanResources(
+                0, rentFee, getLedgerVersion(), mSorobanConfig, mAppConfig,
+                mOpFrame.mParentTx, mDiagnosticEvents))
         {
-            diagnosticEvents.pushError(
-                SCE_BUDGET, SCEC_EXCEEDED_LIMIT,
-                "operation byte-read resources exceeds amount specified",
-                {makeU64SCVal(metrics.mLedgerReadByte),
-                 makeU64SCVal(resources.diskReadBytes)});
-            innerResult(res).code(RESTORE_FOOTPRINT_RESOURCE_LIMIT_EXCEEDED);
-            return {false, {}};
+            innerResult(mRes).code(
+                RESTORE_FOOTPRINT_INSUFFICIENT_REFUNDABLE_FEE);
+            return false;
         }
+        innerResult(mRes).code(RESTORE_FOOTPRINT_SUCCESS);
+        return true;
+    }
+};
 
-        // To maintain consistency with InvokeHostFunction, TTLEntry
-        // writes come out of refundable fee, so only add entrySize
-        metrics.mLedgerWriteByte += entrySize;
-        if (!validateContractLedgerEntry(lk, entrySize, sorobanConfig,
-                                         appConfig, mParentTx,
-                                         diagnosticEvents))
+class RestoreFootprintPreV23ApplyHelper
+    : virtual public RestoreFootprintApplyHelper,
+      virtual public PreV23LedgerAccessHelper
+{
+  public:
+    RestoreFootprintPreV23ApplyHelper(
+        AppConnector& app, AbstractLedgerTxn& ltx, OperationResult& res,
+        std::optional<RefundableFeeTracker>& refundableFeeTracker,
+        OperationMetaBuilder& opMeta, RestoreFootprintOpFrame const& opFrame)
+        : RestoreFootprintApplyHelper(app, res, refundableFeeTracker, opMeta,
+                                      opFrame)
+        , PreV23LedgerAccessHelper(ltx)
+    {
+    }
+
+    std::optional<LedgerEntry>
+    getHotArchiveEntry(LedgerKey const& key) override
+    {
+        // There is no hot archive pre-23.
+        return std::nullopt;
+    }
+    bool
+    entryWasRestored(LedgerKey const& key) override
+    {
+        // NB: even though pre-23 can answer this question precisely, the code
+        // in pre-23 wasn't sensitive to it, so we preserve that functionality.
+        return false;
+    }
+    void
+    restoreEntry(LedgerKey const& lk, LedgerEntry const& entry_,
+                 LedgerKey const& ttlKey,
+                 std::optional<LedgerEntry> const& ttlEntryOpt,
+                 uint32_t restoredLiveUntilLedger) override
+    {
+        // Entry exists in the live BucketList if we get to this point due to
+        // the constTTLLtxe loadWithoutRecord logic above. Get the actual ledger
+        // entry (since we know it exists already at this point)
+        //
+        // NB: this method is given a parameter (entry_) that is the result of
```

Comment:
I think the original reason why I did this is because of some sort of ltx lifetime issue with the constLtxe. I used `getNewestVersion` here instead of `load` because we don't want to update the lastModifiedLedgerSeq of the data/code entry being restored. This was several refactors ago, so we can maintain the `getNewestVersion` here since this is just for historical replay now anyway, but I don't think there's an issue with just piping through the result of the previous loadWithoutRecord, we can just leave it as is though.

Reply:
fixed to use the existing key.